pyuvsim/source.py

```diff
@@ -241,3 +241,41 @@ def get_size(self):
         Estimate the SkyModel memory footprint in bytes
         """
         return self.Ncomponents * self._basesize
+
+
+def read_hdf5(hdf5_healpix_filename, freq_ind):
+    """
+    Read hdf5 files using h5py and get a healpix map, indices and frequencies
+
+    Args:
+        hdf5_filename: path and name of the hdf5 file to read
+        freq_ind: index of the frequency, for which you want to get the healpix map
+    """
+    import h5py
+    f = h5py.File(hdf5_filename)
+    hpmap = f['data'][0, :, freq_ind]
+    indices = f['indices'][()]
+    freqs = f['freqs'][freq_ind] * np.ones(len(indices),)
+    return hpmap, indices, freqs
+
+
+def healpix_to_sky(hpmap, indices, freqs):
+    """
+    Convert a healpix map to a set of point source components at given ICRS ra/dec coordinates, with a
+    flux densities defined by stokes parameters.
+
+    Args:
+        hpmap: healpix map that you get from reading hdf5 file
+        indices: indices from a hdf5 file
+        freqs: frequencies from a hdf5 file
+    """
+    import healpy as hp
+    Nside = hp.npix2nside(hpmap.size)
+    dec, ra = hp.pix2ang(Nside, indices, lonlat=True)
+    dec = Angle(dec, unit='deg')
+    ra = Angle(ra, unit='deg')
+    freq = Quantity(freqs, 'hertz')
+    stokes = np.zeros((4, len(indices)))
+    stokes[0] = hpmap
```

Comment:
We'll need to discuss this some more. We're effectively treating healpix pixels (in Kelvin) as point sources (in Jy), which means we'll need to convert the units along the way. This will depend on the map resolution.